### PR TITLE
Added Java Resources

### PR DESCRIPTION
**Pull request by itsarvindh**
I have added a new section for java resources and added 5 new resource links. Please let me know if I need to make any changes.

This PR is a part of Hacktoberfest 2022.

### CODE REVIEW

**Comment by aaryan2134** [comment]
Please add this section in the index as well

File: README.md [hunk 1]
```diff
@@ -9,6 +9,7 @@ List of resources and project samples for Engineering Students
   - [CS Resources <a name="education"></a>](#cs-resources-)
   - [Machine Learning Resources <a name="machine-learning"></a>](#machine-learning-resources-)
   - [Competitive Programming <a name="competitive-programming"></a>](#competitive-programming-)
+  - [Java Programming <a name="java-programming"></a>](#java-programming)
```
Comment (aaryan2134) [inline README.md:12]: The linking is not right. Not working
Please change it to this
"#java-programming-"

**Review by aaryan2134 — CHANGES_REQUESTED** [review]
Please make this change

File: README.md (hunk 1 above)
Comment (itsarvindh) [reply README.md]: Sorry for the issue, have fixed the issue and recommited ! Please check

**Comment by aaryan2134** [comment]
Looking good.
Please resolve conflicts so that merging can be done

**Comment by itsarvindh** [comment]
Conflict resolved!

**Comment by aaryan2134** [comment]
Thank you for your contribution

**Pull request merged**